src/thumb/op_immediate.rs

```diff
@@ -0,0 +1,76 @@
+use crate::types::{Register, Immediate};
+use crate::get_bits;
+use num_derive::FromPrimitive;    
+use num_traits::FromPrimitive;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, FromPrimitive)]
+pub enum OpImmediateOpCode {
+    MOV = 0,
+    CMP = 1,
+    ADD = 2,
+    SUB = 3
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct OpImmediate {
+    op: OpImmediateOpCode,
+    dest: Register,
+    offset: Immediate
```

Comment:
Does this type really bring us much value? I used just u8 here, I could switch I guess but what other than aliasing u8 does it give us? not every immediate will fill it as well

Reply:
I mean the immediate type